authentication/auth.go

```diff
@@ -527,16 +505,24 @@ func GenerateTokenFromDefaultsV2(schemaURL string, accountServiceURL string, url
 	if claims["survey_metadata"] != nil {
 		surveyMetadata = claims["survey_metadata"].(map[string]interface{})
 	}
-	for key, value := range data {
-		updatedData[key] = value
-	}
 
 	initialData := surveyMetadata["data"].(map[string]interface{})
 
 	for key, value := range initialData {
 		updatedData[key] = value
 	}
 
+	requiredSchemaMetadata = addAdditionalMetadata(surveyMetadata, requiredSchemaMetadata)
+
+	for _, metadata := range requiredSchemaMetadata {
+		if metadata.Validator == "boolean" {
```

Comment:
Are these defaults not generated (is this method not used) only if you go through quick launch? It seems like it's only executed inside this `quickLauncherHandler()` function.